de.metas.dunning/src/main/java/de/metas/dunning/invoice/model/validator/C_Dunning_Candidate.java

```diff
@@ -43,18 +42,35 @@
 			, ifColumnsChanged = I_C_Dunning_Candidate.COLUMNNAME_DunningGrace)
 	public void setDunningGraceOnInvoice(final I_C_Dunning_Candidate candidate)
 	{
-		if (candidate.getAD_Table_ID() != MTable.getTable_ID(I_C_Invoice.Table_Name))
+		if (candidate.getAD_Table_ID() != InterfaceWrapperHelper.getTableId(I_C_Invoice.class))
 		{
 			return;
 		}
-		
+
 		final Properties ctx = InterfaceWrapperHelper.getCtx(candidate);
 		final String trxName = InterfaceWrapperHelper.getTrxName(candidate);
 
 		final I_C_Invoice invoice = InterfaceWrapperHelper.create(ctx, candidate.getRecord_ID(), I_C_Invoice.class, trxName);
 		invoice.setDunningGrace(candidate.getDunningGrace());
-		
 		InterfaceWrapperHelper.setDynAttribute(invoice, POATTR_CallerPO, candidate);
 		InterfaceWrapperHelper.save(invoice);
 	}
+
+	@ModelChange(timings = ModelValidator.TYPE_AFTER_CHANGE
+			, ifColumnsChanged = I_C_Dunning_Candidate.COLUMNNAME_IsDunningDocProcessed)
+	public void setDunningLevelOnInvoice(final I_C_Dunning_Candidate candidate)
+	{
+		if (candidate.getAD_Table_ID() != InterfaceWrapperHelper.getTableId(I_C_Invoice.class)
+				|| !candidate.isDunningDocProcessed())
+		{
+			return;
+		}
+
+		final Properties ctx = InterfaceWrapperHelper.getCtx(candidate);
+		final String trxName = InterfaceWrapperHelper.getTrxName(candidate);
+
+		final I_C_Invoice invoice = InterfaceWrapperHelper.create(ctx, candidate.getRecord_ID(), I_C_Invoice.class, trxName);
```

Comment:
i recomment to instead use `InterfaceWrapperHelper.load() next time

Reply:
Thx!